pkg/framework/objects/service_token/resource.go

```diff
@@ -0,0 +1,325 @@
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/int64default"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/setdefault"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+)
+
+var (
+	_ resource.Resource                = &serviceTokenResource{}
+	_ resource.ResourceWithConfigure   = &serviceTokenResource{}
+	_ resource.ResourceWithImportState = &serviceTokenResource{}
+)
+
+func ServiceTokenResource() resource.Resource {
+	return &serviceTokenResource{}
+}
+
+type serviceTokenResource struct {
+	client *dbt_cloud.Client
+}
+
+// Metadata implements resource.Resource.
+func (st *serviceTokenResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_service_token"
+}
+
+// Configure implements resource.ResourceWithConfigure.
+func (st *serviceTokenResource) Configure(ctx context.Context, req resource.ConfigureRequest, resp *resource.ConfigureResponse) {
+	switch c := req.ProviderData.(type) {
+	case nil: // do nothing
+	case *dbt_cloud.Client:
+		st.client = c
+	default:
+		resp.Diagnostics.AddError("Missing client", "A client is required to configure the service token resource")
+	}
+}
+
+// Schema implements resource.Resource.
+func (st *serviceTokenResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{
+				Computed:    true,
+				Description: "The ID of the service token",
+				// this is used so that we don't show that ID is going to change
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.UseStateForUnknown(),
+				},
+			},
+			"uid": schema.StringAttribute{
+				Description: "Service token UID (part of the token)",
+				Computed:    true,
+			},
+			"name": schema.StringAttribute{
+				Description: "Service token name",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(),
+				},
+			},
+			"token_string": schema.StringAttribute{
+				Description: "Service token secret value (only accessible on creation))",
+				Computed:    true,
+				Sensitive:   true,
+			},
+			"state": schema.Int64Attribute{
+				Description: "Service token state (1 is active, 2 is inactive)",
+				Optional:    true,
+				Computed:    true,
+				Default:     int64default.StaticInt64(1),
+			},
```

Comment:
Not related to this change but it reminds me that I think we should maybe at some point get rid of all the `state` fields. We can't create "deleted" resources or modify them to be "deleted" so the parameter doesn't make sense anywhere.

But, yes, some people might have added it to their configs.

Reply:
I'll create a Task Ticket to make all `state` fields Optional + Deprecated. We can push a release without breaking anyone and give a timeline for when it will be removed.

Reply:
See TFPCLD-13 & TFPCLD-14